allensdk/brain_observatory/behavior/data_objects/running_speed/running_speed.py

```diff
@@ -63,6 +64,32 @@ def __init__(
         self._stimulus_timestamps = stimulus_timestamps
         self._filtered = filtered
 
+    @staticmethod
+    def _check_running_speed(running_speed: pd.DataFrame) -> pd.DataFrame:
+        """Check for negative average running speed.
+
+        If an average < -1 cm/s is found, flip the speed value. This function
+        is needed as a set of experiments for VBN were found to have the
+        polarity of their running speed reversed, making it appear as if the
+        mouse was running backwards.
+
+        Parameters
+        ----------
+        running_speed : pandas.DataFrame
+            Running speed DataFrame.
+
+        Returns
+        -------
+        output_frame : pandas.DataFrame
+           DataFrame with potentially flipped value of running speed if the
+           average speed is < -1 cm/s
+        """
+        mean_speed = running_speed['speed'].mean()
+        output_frame = running_speed.copy()
+        if mean_speed < -1:
```

Comment:
How was it decided that mean_speed < -1 indicates flipped polarity? Why not < 0?

Reply:
I have the same question. Seems like a negative average running speed is bad.

Reply:
This was proposed by the engineering group according to the issue. My assumption the value is -1 as you could have a case where a mouse is not running very much and the noise puts causes the mean to be below zero even fractionally. You can check out the plots of the mean histogram [here](https://github.com/AllenInstitute/AllenSDK/issues/2661). You can see a peak at just above zero with a slight tail to zero and lower values. @corbennett do you have any further context?

Reply:
ok, please link to that histogram in the code so it is clear why that value is hardcoded